public_html/js/index.js

```diff
@@ -49,17 +44,61 @@ $( document ).ready( function () {
 				$reviewerNode.fadeIn( 'slow' );
 			} else if ( ret.error === 'Unauthorized' ) {
 				alert( 'You need to be logged in to be able to review.' );
-				$( buttonId ).addClass( 'btn-' + buttonClass ).removeClass( 'btn-' + buttonClass + '-clicked' ).blur();
-				$( unusedButtonId ).removeClass( 'btn-secondary' ).prop( 'disabled', false ).addClass( 'btn-' + unusedButtonClass );
+				// go back to initial state
+				setReviewState( id, 'open' );
 			} else {
 				alert( 'There was an error in connecting to database.' );
-				$( buttonId ).addClass( 'btn-' + buttonClass ).removeClass( 'btn-' + buttonClass + '-clicked' ).blur();
-				$( unusedButtonId ).removeClass( 'btn-secondary' ).prop( 'disabled', false ).addClass( 'btn-' + unusedButtonClass );
+				setReviewState( id, 'open' );
 			}
+
+			document.activeElement.blur(); // remove focus from button
 		} );
 	}
 
 	/**
+	 * Undo a review
+	 * @param id int ID of the record
+	 * @param oldStatus string current review state of the record
+	 */
+	function undoReview( id, oldStatus ) {
+		// update styles before AJAX to make it seem more responsive
+		setReviewState( id, 'open' );
+
+		$.ajax( {
+			url: 'review/add',
```

Comment:
I think we should do the Undo part with a new route 'review/undo' and write a ReviewUndo controller for it. We can move all the Review controllers in a Review directory. It seems far neater.